public/app/plugins/datasource/grafana-azure-monitor-datasource/components/ResourcePicker/NestedRows.tsx

```diff
@@ -1,66 +1,98 @@
 import { cx } from '@emotion/css';
 import { Checkbox, HorizontalGroup, Icon, IconButton, useStyles2, useTheme2 } from '@grafana/ui';
-import React, { Fragment, useCallback } from 'react';
+import React, { useCallback, useState } from 'react';
 import getStyles from './styles';
-import { EntryType, Row } from './types';
+import { EntryType, Row, RowGroup } from './types';
 
 interface NestedRowsProps {
-  rows: Row[];
+  rows: RowGroup;
   level: number;
-  selected: string[];
-  onRowToggleCollapse: (row: Row) => void;
+  selectedRows: RowGroup;
+  fetchNested: (row: Row) => Promise<void>;
   onRowSelectedChange: (row: Row, selected: boolean) => void;
 }
 
-const NestedRows: React.FC<NestedRowsProps> = ({ rows, selected, level, onRowToggleCollapse, onRowSelectedChange }) => {
+const NestedRows: React.FC<NestedRowsProps> = ({ rows, selectedRows, level, fetchNested, onRowSelectedChange }) => (
+  <>
+    {Object.keys(rows).map((rowId) => (
+      <NestedRow
+        key={rowId}
+        row={rows[rowId]}
+        selectedRows={selectedRows}
+        level={level}
+        fetchNested={fetchNested}
+        onRowSelectedChange={onRowSelectedChange}
+      />
+    ))}
+  </>
+);
+
+interface NestedRowProps {
+  row: Row;
+  level: number;
+  selectedRows: RowGroup;
+  fetchNested: (row: Row) => Promise<void>;
+  onRowSelectedChange: (row: Row, selected: boolean) => void;
+}
+
+const NestedRow: React.FC<NestedRowProps> = ({ row, selectedRows, level, fetchNested, onRowSelectedChange }) => {
   const styles = useStyles2(getStyles);
 
+  const isSelected = !!selectedRows[row.id];
+  const isDisabled = Object.keys(selectedRows).length > 0 && !isSelected;
```

Comment:
why is an entry disabled if it's not selected? (or am I reading this wrong?)

Reply:
Yeah I didn't modify the behavior that was in [the original pr](https://github.com/grafana/grafana/pull/33879/files#diff-166cf4590c5fef3151bb77b3474c4c32ec73d045e7c0a95dcb22a82ff6fba59dR22) just reworked the data structure here. 

I think the idea is to demonstrate that you only can make one selection at a time so it forces you unselect your first choice before you can select another? I also found this a little clunky using it. I think I'd expect to be able to select something and it would auto unselect the other stuff. 

I can change that in this pr if we'd like!